Modules/signalmodule.c

```diff
@@ -148,6 +148,9 @@ get_signal_state(PyObject *module)
 static inline int
 compare_handler(PyObject *func, PyObject *dfl_ign_handler)
 {
+    if (func == NULL || dfl_ign_handler == NULL) {
```

Comment:
It's not really clear to me _why_ this situation could arise but it is a valid check regardless.

Reply:
In my testing it happens when GC runs between `PyErr_SetInterrupt` and  `PyErr_CheckSignals`.